packages/grid/x-data-grid/src/components/containers/GridRootStyles.ts

```diff
@@ -115,6 +115,7 @@ export const GridRootStyles = styled('div', {
     outline: 'none',
     height: '100%',
     display: 'flex',
+    overflow: 'hidden',
```

Comment:
The solution works for displays grid and flex but for table it still [causes](https://codesandbox.io/s/datagrid-6-0-0-in-grid-forked-qgwo7j?file=/demo.tsx) the same issue. I played around and display: grid seems to solve all the cases, do you see any caveats in changing root element's display property from flex to grid 🤔
```suggestion
    display: 'grid',
```

Reply:
Unfortunately `display: grid` causes regressions like this one https://codesandbox.io/s/sad-spence-dv82dy and more (at least 12 unit tests fail).

Thanks for pointing out the issue with `display: table`, I'll look into it!

Reply:
@MBilalShafi 
I think the behavior with `display: table` is expected.
I created this demo with a simple `GridMock` component and I believe the data grid should behave in the same way:
https://codesandbox.io/s/datagrid-6-0-0-in-grid-forked-fb4yjg?file=/demo.tsx

<img width="777" alt="Screenshot 2023-04-22 at 16 38 43" src="https://user-images.githubusercontent.com/13808724/233791093-5865a22e-5cf5-45f3-90fb-7e6d8b219b99.png">

Reply:
Alright, so that's how `display: table` natively works. Thanks for providing the demo.